Wikipedia/Code/Session.swift

```diff
@@ -32,15 +32,15 @@ import Foundation
     }
 
     @objc public static let shared = Session()
-    
-    public override init() {
+
+    private lazy var session: URLSession = {
+        let sharedSession = URLSession.shared
         guard SessionSingleton.sharedInstance().shouldSendUsageReports, let appInstallID = UserDefaults.wmf_userDefaults().wmf_appInstallID else {
-            return
+            return sharedSession
         }
-        session.configuration.httpAdditionalHeaders = ["X-WMF-UUID": appInstallID]
-    }
-
-    fileprivate let session = URLSession.shared
+        sharedSession.configuration.httpAdditionalHeaders = ["X-WMF-UUID": appInstallID]
```

Comment:
is this what you were thinking @joewalsh? 🤔

Reply:
@nambatee yeah, but I think we need to update it if the `shouldSendUsageReports` preference changes